### DISCUSSION TITLE
startup_mtex error

### DISCUSSION THREAD

**Discussion by Patcurran** [mtex-toolbox/mtex#1622] [Ask Anything] [answered]
Hi used mtex around a year ago and just came back to it. My start up mtex wasn't working. So i deleted MTEX and redown-loaded it. My MatLab verson is R2021a academic version. I keep getting the following error: I was including the mtex path but still got the following error:

```
>> startup_mtex
initialize MTEX 5.8.1  ..Unrecognized function or variable 'setMTEXpref'.

Error in startup_mtex (line 54)
setMTEXpref('mtexPath',local_path);
```
is there anyway to solve this?
If you have any advice for me that would be super helpful! :)

**Comment by kilir** [comment]
Hi,
how did you add the mtex path? `addpath` ?
Cheers, 
Rüdoger

**Reply by Patcurran** [reply]
Hi Rüdoger,
I added the path first by hovering over the file, right clicking on the file, hovering over 'add to path', then clicking 'selected folders and subfolders'
then that didn't work so i tired the `addpath` method. Which also gave the same error as before.

My understanding is MTEX has been added to the path, because if it wasn't i would get the following error:
```
>> startup_mtex
Unrecognized function or variable 'startup_mtex'.
```

So i'm not sure if something else is the issue?

cheers,
Patrick

**Comment by zmichels** [comment]
When I do it… I just enter the folder (don’t add it)… then, while inside the folder, run startup_mtex. The install routine will add the proper folders to the path.On Nov 6, 2022, at 3:32 AM, Patcurran ***@***.***> wrote:﻿
Hi Rüdoger,
I added the path first by hovering over the file, right clicking on the file, hovering over 'add to path', then clicking 'selected folders and subfolders'
then that didn't work so i tired the addpath method,
My understanding is MTEX has been added to the path, because if it wasn't i would get the following error:
>> startup_mtex Unrecognized function or variable 'startup_mtex'.
cheers,
Patrick

**Comment by Liujianhua00** [comment]
I meet similar problem before, maybe you can try this. Before you start your Matlab, right-click the shortcut of your Matlab on the desktop, find the shortcut in the property, change your start location to where you put your MTEX, and apply it. Then you can start you Matlab and add the path of MTEX. finally use the code 'startup_mtex' to reinstall MTEX.

**Comment by Patcurran** [comment]
hi both, i tired both of these methods and they didn't work unfortunately. thank you for your suggestions though.

**Comment by ralfHielscher** [comment]
Hi Patcurran,

I'm sorry you have these problems. I would guess that you have not correctly unzip the download. 
The command ```setMTEXpref``` corresponds to the file ```mtex/option_tools/setMTEXpref.m```.
Could you please check that this file does exist on your computer.
You can also check whether Matlab can find this file using the command
```
which setMTEXpref
```
I hope you can get MTEX running for you.
Ralf.

**Comment by Patcurran** [comment]
Hi Ralf,

I unzipped the folder with the 'extract' button as described [here ](https://support.microsoft.com/en-us/windows/zip-and-unzip-files-f6dde0a7-0fec-8294-e1d3-703ed85e7ebc#:~:text=To%20unzip%20files,folder%20to%20a%20new%20location.)

i do not have this file `mtex/option_tools/setMTEXpref.m` or even the `option_tools` folder.

the result from `which setMTEXpref` was this:
```
>> which setMTEXpref
'setMTEXpref' not found.
```

I'm not sure whats gone wrong.

![image](https://user-images.githubusercontent.com/78144670/200341408-1ade7ff5-64c3-471c-8501-165aa20c4424.png)


Thanks for your help.
Pat

**Comment by ralfHielscher** [comment, marked as answer]
Maybe you download the zip ones again. For some reason it happens sometimes, that the zip is not completely downloaded and some directories are missing. I have just checked it myself. The required directory is included in the zip file available at the Download page. Your zip is missing many more directories.

All the best,
Ralf.

**Reply by Patcurran** [reply]
Hi Ralf,

Thank you for your help.
I re-downloaded it (this is now the 4th time) and i got the same result. Do you have any suggestions?

best wishes,
Patrick

**Reply by ralfHielscher** [reply]
Hi Patrick,

I have put the uncompressed toolbox on https://ificloud.xsitepool.tu-freiberg.de/index.php/s/om6jXBydNSoYGX7
and the compressed toolbox https://ificloud.xsitepool.tu-freiberg.de/index.php/s/EYSCPTFLEZrJ7tH
at a different download location. Maybe this works better for you.

Ralf.

**Reply by Patcurran** [reply]
thank you very much Ralf that is working. I was using google chrome, i was unable to download the compressed file you have shared here. I switched to Microsoft edge and now it has worked and i have opened up MTEX on my PC! :) 

thanks for all the help everyone.
Pat

**Comment by Liujianhua00** [comment]
My Matlab verson is also R2021a, when I start my Matlab, it will automatically install the MTEX as the screenshot displays. I think the problem may not be the zip file you downloaded. The unzipped MTEX folder should be included in the Matlab folder for certain, and add the path of MTEX folder and subfolders before you install the MTEX as you did before.
![image](https://user-images.githubusercontent.com/114222474/200537174-bdcca758-6fca-48f0-970f-26f7e2c154ce.png)

**Reply by Patcurran** [reply]
thank you, i have it sorted now :)
